### DISCUSSION TITLE
Path Directory Not Found

### DISCUSSION THREAD

**Discussion by keshav12-deloitte** [community/community#39661] [Other Feature Feedback, Questions, & Ideas]
Select Topic Area

Question

### Body

[Errno 2] No such file or directory .I am facing this error when in try to run my python file it says python file not found .Can anyone share from which directory i should give if my python file is present in Tests Folder.
Any Answer related to path is appreciated.

I want to run name.py file can anyone help me how to set the path
![Screenshot (322)](https://user-images.githubusercontent.com/101085487/203234495-10e8149d-29ba-4627-9410-8bd6dc35fbad.png)

**Comment by DSDmark** [comment]
Hi @keshav12-deloitte,

Can you **specific more** about the details that you want? For better understanding 😅. Also, you can add some **screenshots** or **errors** that you were facing 👍.

Is hard to tell about your error. Also, should add your path detail if your error about that path. I mean add screenshot your path.

**Reply by keshav12-deloitte** [reply]
@DSDmark here are the screenshots of my project structure could you help me out how can i run name.py to run

**Reply by DSDmark** [reply]
Hi @keshav12-deloitte,

just download your that **name.py** in your local system. After that, open your **cmd** by click right like this.

![2022-11-22 (2)](https://user-images.githubusercontent.com/85744816/203238793-cc3b5446-96d5-4c7f-8630-6b72dcff6e05.png)

And After that run that command ```python name.py``` like that.

![2022-11-22 (3)](https://user-images.githubusercontent.com/85744816/203238916-6c592809-327e-436e-85f6-6b8cf98733a0.png)

**Reply by Som23Git** [reply]
Hi @keshav12-deloitte,

I could see that @DSDmark had shared a solution to run the `python` script `name.py`. Can you kindly highlight the details of the environment in which you're trying to run this `name.py`? Because I could see that there is a main `YML` file where it creates a cron job to send the commit/updates to the repo and it runs through the python script. 
> Please note: If you're using containers then you should install packages within the containers because the containers are isolated from the main host/kernel so it cannot find the path for the `Python` package.

**Comment by david-engelmann** [comment]
@keshav12-deloitte It seems that your python interpreter is not configured correctly. [Here is some good documentation about setting up the interpreter](https://docs.python.org/3/tutorial/interpreter.html). If the results of `whereis python` or `whereis python3` looks correct then it's possible that the issue is calling the test file. Depending on the results of `pwd` you'll need to adjust the path to the python test file accordingly. I'd add an example file system below as an example

|-project
|--src
|---src_file.py
|--test
|---test_file.py

If `pwd` returns blah/project, then you'll want to call the python file with the following command `python3 test/test_file.py`

if `pwd` returns blah/project/test, then you'll want to call the python file with the following command `python3 test_file.py`

if `pwd` returns blah/project/src, then you'll want to call the python file with the following command `python3 ../test/test_file.py`
